compiler/rustc_smir/src/stable_mir/ty.rs

```diff
@@ -242,3 +242,35 @@ pub struct BoundTy {
     pub var: usize,
     pub kind: BoundTyKind,
 }
+
+pub type Bytes = Vec<u8>;
+pub type Size = usize;
+pub type Prov = usize;
```

Comment:
This should be `AllocId`. How is that exposed to smir?

Reply:
Similar to how we have a table for `DefId`s, we should add something like it for `AllocId`s (instead of exposing the `AllocId`s internals). Since we don't actually need this in this PR yet, we can make `Prov` just be `Opaque` and stick a stringified version of the `AllocId` in it.

Reply:
I made it `Opaque`